File: cases/taint/php/return-relay-two-hop-positive/return_relay_two_hop_positive.php
<?php
function dfb_source(): string { // DFB-SOURCE: return-two-hop-input
    return "tainted";
}

function firstRelay(string $value): string { // DFB-WITNESS: return-two-hop-first
    return $value;
}

function secondRelay(string $value): string { // DFB-WITNESS: return-two-hop-second
    return firstRelay($value);
}

function dfb_sink(string $value): void {} // DFB-SINK: return-two-hop-sink

function run(): void {
    $result = secondRelay(dfb_source());
    dfb_sink($result);
}

File: cases/taint/rust/return-relay-two-hop-positive/return_relay_two_hop_positive.rs
fn dfb_source() -> i32 { // DFB-SOURCE: return-two-hop-input
    1
}

fn first_relay(value: i32) -> i32 { // DFB-WITNESS: return-two-hop-first
    value
}

fn second_relay(value: i32) -> i32 { // DFB-WITNESS: return-two-hop-second
    first_relay(value)
}

fn dfb_sink(value: i32) {} // DFB-SINK: return-two-hop-sink

fn run() {
    let result = second_relay(dfb_source());
    dfb_sink(result);
}
